testsuite/tests/init/with-chdir/test.py

```diff
@@ -0,0 +1,20 @@
+"""
+Test "executable" only appears in --bin initializations
+"""
+
+import os.path
+
+from drivers.alr import run_alr
+from drivers.asserts import assert_match
+from drivers.helpers import content_of
+
+test_dir = os.getcwd()
+
+# Binary crate
+run_alr("init", "--bin", "xxx")
+assert_match(".*executables = \[", content_of("xxx/alire.toml"))
+
+# Check that it builds and runs
+run_alr("--chdir=xxx", "run")
```

Comment:
Also, it would be good to have a check after this one that when the target directory doesn't exist, alr exits with error.